### ISSUE TITLE
[FEATURE] Display `link-name` flag even if optional

### ISSUE THREAD

**Issue by brooksmtownsend** [wasmCloud/wash-archive#247]
**Is your feature request related to a problem? Please describe.**
When starting capability providers or putting link definitions, the `link-name` flag is "optional" because it can default to the value `default`. This works well in development environments where the link name is rarely modified, but in more complex scenarios link names are used frequently. The value of `default` is reasonable as it covers the 80%+ use case, but in the help text for `wash` we should show `--link-name` as a flag.

Today, this is what `wash ctl link put` shows:
```shell
~/github.com/cosmonic/demo-scripts ➜ wash ctl link put
error: The following required arguments were not provided:
    <actor-id>
    <provider-id>
    <contract-id>

USAGE:
    wash ctl link put [OPTIONS] <actor-id> <provider-id> <contract-id> [values]...

For more information try --help
```

**Describe the solution you'd like**
Ideally, the above message should be:
```shell
~/github.com/cosmonic/demo-scripts ➜ wash ctl link put
error: The following required arguments were not provided:
    <actor-id>
    <provider-id>
    <contract-id>

USAGE:
    wash ctl link put --link-name <LINK_NAME> [OPTIONS] <actor-id> <provider-id> <contract-id> [values]...

For more information try --help
```
The difference here is that the `--link-name` flag is shown even though it is optional. There should be a `clap` option to do this.

**Describe alternatives you've considered**
Alternatively we could remove the default value of `default`, but this would introduce a jarring breaking change and negatively affect the 80% use case.

**Comment by emattiza** [comment]
i took a look at this and only found a clap option for override_usage and hard coding the usage string. Thoughts? Have a branch i can bring over for review.

**Comment by brooksmtownsend** [comment]
@emattiza yeah would love to take a look at what that looks like, I haven't had a chance to look at the clap args quite yet but feel free to open a draft PR and we can collaborate there

**Comment by brooksmtownsend** [comment]
Fixed in #254
